### PR TITLE
[FLINK-31876][QS] Migrate flink-queryable-state-client tests to JUnit5

### PR DESCRIPTION
**Pull request by fredia**
## What is the purpose of the change

Migrate flink-queryable-state-client testsD to JUnit5

### CODE REVIEW

File: flink-queryable-state/flink-queryable-state-client-java/src/test/java/org/apache/flink/queryablestate/client/state/ImmutableListStateTest.java [hunk 1]
```diff
@@ -56,26 +57,34 @@ public void setUp() throws Exception {
...
-    @Test(expected = UnsupportedOperationException.class)
-    public void testUpdate() throws Exception {
-        List<Long> list = getStateContents();
-        assertEquals(1L, list.size());
+    @Test
+    void testUpdate() {
+        assertThatThrownBy(
+                        () -> {
+                            List<Long> list = getStateContents();
+                            assertEquals(1L, list.size());
```
Comment (jiexray) [inline flink-queryable-state/flink-queryable-state-client-java/src/test/java/org/apache/flink/queryablestate/client/state/ImmutableListStateTest.java:65]: Better use `assertThat(list).hasSize(1)`

File: flink-queryable-state/flink-queryable-state-client-java/src/test/java/org/apache/flink/queryablestate/client/state/ImmutableListStateTest.java [hunk 2]
```diff
@@ -56,26 +57,34 @@ public void setUp() throws Exception {
...
 
-        long element = list.get(0);
-        assertEquals(42L, element);
+                            long element = list.get(0);
+                            assertEquals(42L, element);
```
Comment (jiexray) [inline flink-queryable-state/flink-queryable-state-client-java/src/test/java/org/apache/flink/queryablestate/client/state/ImmutableListStateTest.java:68]: better use `assertThat(list).contains(42L, Index.atIndex(0));`

File: flink-queryable-state/flink-queryable-state-client-java/src/test/java/org/apache/flink/queryablestate/client/state/ImmutableListStateTest.java [hunk 3]
```diff
@@ -56,26 +57,34 @@ public void setUp() throws Exception {
...
-    @Test(expected = UnsupportedOperationException.class)
-    public void testClear() throws Exception {
-        List<Long> list = getStateContents();
-        assertEquals(1L, list.size());
+    @Test
+    void testClear() {
+        assertThatThrownBy(
+                        () -> {
+                            List<Long> list = getStateContents();
+                            assertEquals(1L, list.size());
```
Comment (jiexray) [inline flink-queryable-state/flink-queryable-state-client-java/src/test/java/org/apache/flink/queryablestate/client/state/ImmutableListStateTest.java:80]: better use `assertThat(list).hasSize(1)`

File: flink-queryable-state/flink-queryable-state-client-java/src/test/java/org/apache/flink/queryablestate/client/state/ImmutableListStateTest.java [hunk 4]
```diff
@@ -56,26 +57,34 @@ public void setUp() throws Exception {
...
 
-        long element = list.get(0);
-        assertEquals(42L, element);
+                            long element = list.get(0);
+                            assertEquals(42L, element);
```
Comment (jiexray) [inline flink-queryable-state/flink-queryable-state-client-java/src/test/java/org/apache/flink/queryablestate/client/state/ImmutableListStateTest.java:83]: better use `assertThat(list).contains(42L, Index.atIndex(0));`

File: flink-queryable-state/flink-queryable-state-client-java/src/test/java/org/apache/flink/queryablestate/client/state/ImmutableAggregatingStateTest.java [hunk 5]
```diff
@@ -53,20 +54,28 @@ public void setUp() throws Exception {
...
-    @Test(expected = UnsupportedOperationException.class)
-    public void testUpdate() throws Exception {
-        String value = aggrState.get();
-        assertEquals("42", value);
+    @Test
+    void testUpdate() {
+        assertThatThrownBy(
+                        () -> {
+                            String value = aggrState.get();
+                            assertEquals("42", value);
```
Comment (reswqa) [inline flink-queryable-state/flink-queryable-state-client-java/src/test/java/org/apache/flink/queryablestate/client/state/ImmutableAggregatingStateTest.java:62]: All assertions should be migrated to AssertJ.

File: flink-queryable-state/flink-queryable-state-client-java/src/test/java/org/apache/flink/queryablestate/client/state/ImmutableAggregatingStateTest.java [hunk 6]
```diff
@@ -53,20 +54,28 @@ public void setUp() throws Exception {
...
 
-        aggrState.add(54L);
+                            aggrState.add(54L);
```
Comment (reswqa) [inline flink-queryable-state/flink-queryable-state-client-java/src/test/java/org/apache/flink/queryablestate/client/state/ImmutableAggregatingStateTest.java:64]: We should make the code block of `assertThatThrownBy` as small as possible, ideally it only contains codes that will throw errors.

**Review by reswqa — COMMENTED** [review]
Thanks @fredia for the migrate, I left some comments, PTAL.

File: flink-queryable-state/flink-queryable-state-client-java/src/test/java/org/apache/flink/queryablestate/client/state/ImmutableAggregatingStateTest.java (hunk 5 above)
Comment (fredia) [reply flink-queryable-state/flink-queryable-state-client-java/src/test/java/org/apache/flink/queryablestate/client/state/ImmutableAggregatingStateTest.java:62]: Fixed, replaced `org.junit.Assert.assertEquals` with `org.junit.jupiter.api.Assertions.assertEquals`

**Comment by fredia** [comment]
@jiexray @reswqa Thanks for the review, I updated the PR according to your suggestions, please help take a look again.

File: flink-queryable-state/flink-queryable-state-client-java/src/test/java/org/apache/flink/queryablestate/client/state/ImmutableAggregatingStateTest.java [hunk 7]
```diff
@@ -24,23 +24,24 @@
...
-import org.junit.Before;
-import org.junit.Test;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
 
 import java.io.ByteArrayOutputStream;
 
-import static org.junit.Assert.assertEquals;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertEquals;
```
Comment (reswqa) [inline flink-queryable-state/flink-queryable-state-client-java/src/test/java/org/apache/flink/queryablestate/client/state/ImmutableAggregatingStateTest.java:33]: Why is it still there, we shouldn't use any non-assertJ assertions anymore.

File: flink-queryable-state/flink-queryable-state-client-java/src/test/java/org/apache/flink/queryablestate/client/state/ImmutableListStateTest.java [hunk 8]
```diff
@@ -56,26 +58,23 @@ public void setUp() throws Exception {
...
+    void testUpdate() throws Exception {
         List<Long> list = getStateContents();
-        assertEquals(1L, list.size());
-
-        long element = list.get(0);
-        assertEquals(42L, element);
-
-        listState.add(54L);
+        assertThat(list).hasSize(1);
+        assertThat(list).contains(42L, Index.atIndex(0));
```
Comment (reswqa) [inline flink-queryable-state/flink-queryable-state-client-java/src/test/java/org/apache/flink/queryablestate/client/state/ImmutableListStateTest.java:65]:
```suggestion
    assertThat(list).containsExactly(42L);
```

File: flink-queryable-state/flink-queryable-state-client-java/src/test/java/org/apache/flink/queryablestate/client/state/ImmutableMapStateTest.java [hunk 9]
```diff
@@ -24,28 +24,29 @@
...
 import java.util.Iterator;
 import java.util.Map;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertTrue;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
```
Comment (reswqa) [inline flink-queryable-state/flink-queryable-state-client-java/src/test/java/org/apache/flink/queryablestate/client/state/ImmutableMapStateTest.java:37]: These should all be removed.

File: flink-queryable-state/flink-queryable-state-client-java/src/test/java/org/apache/flink/queryablestate/client/state/ImmutableAggregatingStateTest.java (hunk 7 above)
Comment (fredia) [reply flink-queryable-state/flink-queryable-state-client-java/src/test/java/org/apache/flink/queryablestate/client/state/ImmutableAggregatingStateTest.java]: Sorry for the misunderstanding, replace all assertions of junit5 to assertJ.

File: flink-queryable-state/flink-queryable-state-client-java/src/test/java/org/apache/flink/queryablestate/client/state/ImmutableListStateTest.java [hunk 10]
```diff
@@ -56,26 +57,23 @@ public void setUp() throws Exception {
...
+    void testUpdate() throws Exception {
         List<Long> list = getStateContents();
-        assertEquals(1L, list.size());
-
-        long element = list.get(0);
-        assertEquals(42L, element);
-
-        listState.add(54L);
+        assertThat(list).hasSize(1);
+        assertThat(list).containsExactly(42L);
```
Comment (reswqa) [inline flink-queryable-state/flink-queryable-state-client-java/src/test/java/org/apache/flink/queryablestate/client/state/ImmutableListStateTest.java:64]:
```suggestion
    assertThat(list).containsExactly(42L);
```

File: flink-queryable-state/flink-queryable-state-client-java/src/test/java/org/apache/flink/queryablestate/client/state/ImmutableListStateTest.java [hunk 11]
```diff
@@ -56,26 +57,23 @@ public void setUp() throws Exception {
...
+    void testClear() throws Exception {
         List<Long> list = getStateContents();
-        assertEquals(1L, list.size());
-
-        long element = list.get(0);
-        assertEquals(42L, element);
+        assertThat(list).hasSize(1);
 
-        listState.clear();
+        assertThat(list).containsExactly(42L);
```
Comment (reswqa) [inline flink-queryable-state/flink-queryable-state-client-java/src/test/java/org/apache/flink/queryablestate/client/state/ImmutableListStateTest.java:74]: assertThat(list).containsExactly(42L);

File: flink-queryable-state/flink-queryable-state-client-java/src/test/java/org/apache/flink/queryablestate/client/state/ImmutableMapStateTest.java [hunk 12]
```diff
@@ -63,130 +62,147 @@ public void setUp() throws Exception {
...
+        assertThat(mapState.contains(2L)).isTrue();
         value = mapState.get(2L);
-        assertEquals(5L, value);
+        assertThat(value).isEqualTo(5L);
 
-        mapState.clear();
+        assertThatThrownBy(
+                        () -> {
+                            mapState.clear();
+                        })
```
Comment (reswqa) [inline flink-queryable-state/flink-queryable-state-client-java/src/test/java/org/apache/flink/queryablestate/client/state/ImmutableMapStateTest.java:200]:
```suggestion
                assertThatThrownBy(() -> mapState.clear())
```

**Review by reswqa — COMMENTED** [review]
Thanks @fredia for the update, I only left some minor comment.

**Comment by fredia** [comment]
@reswqa Thanks for the detailed review, fixed as suggested, please help take a look again.

**Review by reswqa — APPROVED** [review]
Thanks @fredia, LGTM % CI.

**Comment by fredia** [comment]
My local CI https://dev.azure.com/fredia/flink/_build/results?buildId=518&view=results has passed.

**Comment by reswqa** [comment]
> My local CI https://dev.azure.com/fredia/flink/_build/results?buildId=518&view=results has passed.

I have squashed and rebased this to origin/master to see if it can pass CI.

**Pull request merged**